src/OrleansRuntime/Streams/QueueBalancer/DeploymentBasedQueueBalancer.cs

```diff
@@ -87,6 +85,8 @@ public class DeploymentBasedQueueBalancer : QueueBalancerBase, ISiloStatusListen
             }
             this.allQueues = new ReadOnlyCollection<QueueId>(queueMapper.GetAllQueues().ToList());
             this.siloMaturityPeriod = siloMaturityPeriod;
```

Comment:
Queue balancer uses the shared state, rather than being  a listener itself, limiting it to only accessing the silo status changes from it's own thread.  In this case, the queue manager system target's thread.